src/mono/mono/mini/decompose.c

```diff
@@ -1538,17 +1538,17 @@ mono_decompose_array_access_opts (MonoCompile *cfg)
 					break;
 				case OP_LDLEN:
 					NEW_LOAD_MEMBASE_FLAGS (cfg, dest, OP_LOADI4_MEMBASE, ins->dreg, ins->sreg1,
-											ins->inst_imm, ins->flags);
+											GINT32_TO_UINT32(ins->inst_imm), ins->flags);
```

Comment:
I don't understand this.  If  I follow the macros, this ends up in `NEW_LOAD_MEMBASE` which does:
```c
		(dest)->inst_offset = (offset); \
```

`inst_offset` is 

```c
#define inst_offset data.op[0].const_val
```

and `const_val` is

```c
			target_mgreg_t const_val;
```

and finally

```c
#if SIZEOF_REGISTER == 4
typedef gint32 target_mgreg_t;
#else
typedef gint64 target_mgreg_t;
#endif
```

So the offset value is signed and gets stored in a signed field.  Why is this cast necessary?

Reply:
The warning happened here:

Reply:
I didn't notice that the size of `target_mgreg_t` could change. If the offset could be negative, I could cast the return of `mono_target_pagesize` to be signed.

Reply:
Yea I'd prefer if we allow the callers of the various MEMBASE macros to pass signed inputs and do the cast on the comparison.

Reply:
Will cast the return of `mono_target_pagesize`